gladier/managers/flows_manager.py

```diff
@@ -337,36 +338,45 @@ def purge_flow(self):
-        Start a Globus Automate flow. By default, the flow definiton is checked and synced if it
-        has changed locally or deployed if it does not exist.
-
-        If a group is set, run permissions are updated and applied to the run (includes
-        'run_managers', 'run_monitors').
-
-        Any scope changes required post-deployment/update are propogated through the login_manager
-        and may require an additional login. A new flow checksum/id may be tracked in storage if
-        the flow changed or was newly deployed.
-
-        :param flow_input: A dict of input to be passed to the automate flow. self.check_input()
-                           is called on each tool to ensure basic needs are met for each.
-                           Input MUST be wrapped inside an 'input' dict,
-                           for example {'input': {'foo': 'bar'}}.
-
-        :param use_defaults: Use the result of self.get_input() to populate base input for the
-                             flow. All conflicting input provided by flow_input overrides
-                             values set in use_defaults.
-        :param \**flow_kwargs: Set several keyed arguments that include the label to be used
-                               in the automate app. If no label is passed the standard automate
-                               label is used. Also ensure label <= 64 chars long.
+    def run_flow(self, flow_input: t.Mapping,
+                 run_managers: t.Optional[t.Iterable[str]] = None,
+                 run_monitors: t.Optional[t.Iterable[str]] = None,
+                 dry_run: bool = False,
+                 label: t.Optional[str] = None,
+                 tags: t.Optional[t.List[str]] = None,
+                 **kwargs):
+        r"""Start a Globus Automate flow. By default, the flow definiton is
+        checked and synced if it has changed locally or deployed if it does not
+        exist.
+
+        If a group is set, run permissions are updated and applied to the run
+        (includes 'run_managers', 'run_monitors').
+
+        Any scope changes required post-deployment/update are propogated through
+        the login_manager and may require an additional login. A new flow
+        checksum/id may be tracked in storage if the flow changed or was newly
+        deployed.
+
+        :param flow_input: A dict of input to be passed to the automate
+                           flow. self.check_input() is called on each tool to
+                           ensure basic needs are met for each.  Input MUST be
+                           wrapped inside an 'input' dict, for example {'input':
+                           {'foo': 'bar'}}.
+
+
+        :param \**flow_kwargs: Set several keyed arguments that include the
+                               label to be used in the automate app. If no label
+                               is passed the standard automate label is
+                               used. Also ensure label <= 64 chars long.
+
```

Comment:
In the other PR, I simply removed the docs since this was more of an internal method. We could add them back here for the time being. This is another place where I'm torn between documenting things and making things DRY.